### PR TITLE
feat(FOROME-950): add styled checkbox

### PR DESCRIPTION
**Pull request by Lapk1n**
PR into https://github.com/ForomePlatform/Anfisa-React-Client/pull/586

https://quantori.atlassian.net/browse/FOROME-950

In addition:
1. Fix unchecking all/not mods on dtree modals

### CODE REVIEW

File: src/ui/check-input/check-input.tsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+      <input
+        id={`${'checkbox'} + ${id}`}
+        type="checkbox"
+        checked={checked}
+        onChange={onChange}
+        className={(cn('hidden'), styles.checkbox)}
+        disabled={disabled}
+      />
+
+      <label
```
Comment (QSdmitrioul) [inline src/ui/check-input/check-input.tsx]: It is seems weird, why you don't custom styles of input, you will able to use input inside label, so you can rid of `id` everywhere. Moreover you styles can be automatically applied for each input with type `checkbox`

**Review by QSdmitrioul — CHANGES_REQUESTED** [review]

**Review by QSdmitrioul — APPROVED** [review]

File: src/ui/check-input/check-input.module.css [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+.checkmark {
+    position: relative;
+    display: block;
+    width: 16px;
+    height: 16px;
+    border: 1px solid theme('colors.grey.blue');
+    border-radius: 0.125rem;
+    transition: all 0.2s;
+    cursor: pointer;
```
Comment (AndreyUstyumenko) [inline src/ui/check-input/check-input.module.css:28]: css modules supports nesting, so &::after {...} here

File: src/ui/check-input/check-input.module.css [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    top: 45%;
+    left: 50%;
+    width: 5px;
+    height: 9px;
+    transform: translate(-50%, -50%) rotateZ(40deg);
+    border-right: 1px solid transparent;
+    border-bottom: 1px solid transparent;
+}
+
+.checkbox:checked + .checkmark {
```
Comment (AndreyUstyumenko) [inline src/ui/check-input/check-input.module.css]: and here below they should be nested too

File: src/pages/filter/ui/selected-group-item.tsx [hunk 4]
```diff
@@ -18,15 +18,19 @@ export const SelectedGroupItem = ({
     handleCheckGroupItem(event.target.checked, variant)
   }
 
+  const variantName = variant[0]
+  const variantValue = variant[1]
```
Comment (AndreyUstyumenko) [inline src/pages/filter/ui/selected-group-item.tsx]: const [variantName, variantValue] = variant

File: src/pages/ws/ui/control-panel/zone-modals/components/zone-modal-list.tsx [hunk 5]
```diff
@@ -33,23 +33,22 @@ export const ZoneModalList = observer(
...
+            (isSamples && zoneStore.localSamples.includes(itemName)) ||
+            (isTags && zoneStore.localTags.includes(itemName))
           return (
             <Checkbox
-              key={item}
-              checked={checked}
-              onChange={e => handleCheck(e.target.checked, item)}
-              className="flex items-center mb-4 text-12"
+              key={itemName + index}
+              checked={checked || false}
```
Comment (AndreyUstyumenko) [inline src/pages/ws/ui/control-panel/zone-modals/components/zone-modal-list.tsx]: checked is boolean already

File: src/pages/filter/ui/modal-edit/components/modal-compound-het/modal-compound-het.store.ts [hunk 6]
```diff
@@ -27,6 +27,16 @@ class ModalCompoundHetStore {
   }
 
   public setCurrentMode(modeType: ModeTypes) {
+    if (!modeType) {
```
Comment (AndreyUstyumenko) [inline src/pages/filter/ui/modal-edit/components/modal-compound-het/modal-compound-het.store.ts]: if modeType can be undefined why it's not optional then?

File: src/pages/filter/ui/modal-edit/components/modal-compound-het/modal-compound-het.store.ts [hunk 7]
```diff
@@ -27,6 +27,16 @@ class ModalCompoundHetStore {
...
+      this.currentMode = undefined
+    }
+
+    if (this.currentMode === modeType) {
```
Comment (AndreyUstyumenko) [inline src/pages/filter/ui/modal-edit/components/modal-compound-het/modal-compound-het.store.ts]: something is wrong here. `this.resetCurrentMode()` does the same thing that you do on 31 line. So if you set mode to undefined, you don't need to call `resetCurrentMode`. Maybe `if (!modeType || this.currentMode === modeType)` ?

File: src/pages/ws/ui/control-panel/zone-modals/components/zone-modal-mods.tsx [hunk 8]
```diff
@@ -23,7 +23,8 @@ export const ZoneModalMods = observer((): ReactElement => {
     <Fragment>
       <div className="flex my-2">
         <Checkbox
-          className="mr-6 flex items-center text-12"
+          id={'zone-not-mod'}
```
Comment (AndreyUstyumenko) [inline src/pages/ws/ui/control-panel/zone-modals/components/zone-modal-mods.tsx]: mode?

File: src/pages/ws/ui/control-panel/zone-modals/components/zone-modal-list.tsx (hunk 5 above)
Comment (Lapk1n) [reply src/pages/ws/ui/control-panel/zone-modals/components/zone-modal-list.tsx]: boolean | undefined

Comment (AndreyUstyumenko) [reply src/pages/ws/ui/control-panel/zone-modals/components/zone-modal-list.tsx]: !!checked then

**Review by AndreyUstyumenko — APPROVED** [review]

**Pull request merged**